pinc/project_states.inc

```diff
@@ -26,90 +26,80 @@ include_once($relPath.'site_vars.php');
 class ProjectState
 {
     public function __construct(
+        $name,
         $medium_label,
-        $long_label,
+        $label,
         $forum,
         $phase,
         $star_metal
     ) {
+        $this->name = $name;
         $this->medium_label = $medium_label;
-        $this->long_label = $long_label;
+        $this->label = $label;
         $this->forum = $forum;
         $this->phase = $phase;
         $this->star_metal = $star_metal;
     }
 }
 
-global $project_states;
+class ProjectStates
+{
+    private static $states = [];
 
-function declare_project_state(
-    $constant_name,
-    $constant_value,
-    $medium_label,
-    $long_label,
-    $forum,
-    $phase,
-    $star_metal
-) {
-    global $project_states;
+    public static function add_state($state)
```

Comment:
So we pass a reference to the state instead of a copy of it. We do this for Activities too.
```suggestion
    public static function add_state(&$state)
```
This is mostly important for when we pass this back to callers -- everyone is working with the same object.

Reply:
I guess it can't hurt.
Or you could avoid using the add_state() function by making $states public and say
```php
    ProjectStates::$states[PROJ_NEW] = new ProjectState( ...
```
or keep $states private and define everything inside ProjectStates using a public function.

Reply:
Done this.